src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -0,0 +1,507 @@
+import { AfterViewInit, Component, ElementRef, Input, OnDestroy, ViewChild, ViewEncapsulation, HostListener } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { fromEvent, Subscription } from 'rxjs';
+
+import { ContentType, LinkType, Orientation, TourStep } from './guided-tour.constants';
+import { GuidedTourService } from './guided-tour.service';
+import { AccountService } from 'app/core';
+
+@Component({
+    selector: 'jhi-guided-tour',
+    templateUrl: './guided-tour.component.html',
+    styleUrls: ['./guided-tour.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+})
+export class GuidedTourComponent implements AfterViewInit, OnDestroy {
+    @ViewChild('tourStep', { static: false }) public tourStep: ElementRef;
+
+    // Used to adjust values to determine scroll. This is a blanket value to adjust for elements like nav bars.
+    public topOfPageAdjustment = 0;
+    // Sets the width of all tour step elements.
+    public tourStepWidth = 500;
+    // Sets the minimal width of all tour step elements.
+    public minimalTourStepWidth = 400;
+    // Sets the highlight padding around the selected .
+    public highlightPadding = 4;
+
+    public currentTourStep: TourStep | null;
+    public selectedElementRect: DOMRect | null;
+
+    private resizeSubscription: Subscription;
+    private scrollSubscription: Subscription;
+
+    readonly ContentType = ContentType;
+    readonly LinkType = LinkType;
+
+    constructor(public sanitizer: DomSanitizer, public guidedTourService: GuidedTourService, public accountService: AccountService) {}
+
+    /**
+     * Enable tour navigation with left and right keyboard arrows and escape key
+     * @param event: keyboard keydown event
+     */
+    @HostListener('document:keydown', ['$event'])
+    handleKeyboardEvent(event: KeyboardEvent) {
+        if (event.code === 'ArrowRight' && this.guidedTourService.currentTourStepDisplay <= this.guidedTourService.currentTourStepCount) {
```

Comment:
Please use if else or switch here, so that other cases are not checked if first case is true.

Reply:
@maxr96 I changed that :)